tests/config/loader/ini/replace/test_replace.py

```diff
@@ -2,26 +2,79 @@
 
 import pytest
 
-from tox.config.loader.ini.replace import find_replace_part
+from tests.config.loader.ini.replace.conftest import ReplaceOne
+from tox.config.loader.ini.replace import MatchExpression, find_replace_expr
+from tox.report import HandledError
 
 
 @pytest.mark.parametrize(
-    ("value", "result"),
+    ("value", "exp_output"),
     [
-        ("[]", (0, 1, "posargs")),
-        ("123[]", (3, 4, "posargs")),
-        ("[]123", (0, 1, "posargs")),
-        (r"\[\] []", (5, 6, "posargs")),
-        (r"[\] []", (4, 5, "posargs")),
-        (r"\[] []", (4, 5, "posargs")),
-        ("{foo}", (0, 4, "foo")),
-        (r"\{foo} {bar}", (7, 11, "bar")),
-        ("{foo} {bar}", (0, 4, "foo")),
-        (r"{foo\} {bar}", (7, 11, "bar")),
-        (r"{foo:{bar}}", (5, 9, "bar")),
-        (r"{\{}", (0, 3, r"\{")),
-        (r"{\}}", (0, 3, r"\}")),
+        ("[]", [MatchExpression([["posargs"]])]),
```

Comment:
Oh my, because you've changed the tests makes this pr infinitely harder to review. How am I supposed to know you did not introduce a lot of breaking changes 😅

Reply:
i'm realizing now that `find_replace_part` was actually a public API of the `tox.config.loader.ini.replace` module, so I don't know if I can just "replace" it like this (even though the only caller in setenv was updated).

I took some time tonight to try to reimplement the `find_replace_part` API on top of the new `find_replace_expr` recursive implementation, and it's definitely possible, but the `MatchExpression` needs to track position information for where in the original value the match was found to allow reconstruction of the`(start, end, replacement)` tuple.

with that shim in place, I'd be able to bring back the original tests with no modification (only new tests, or those with clear behavior changes) to hopefully improve the reviewability here.

the other option would be keeping the regex parser, or at least the `find_replace_expr` part as it is currently and leave these tests in place... even though the ini-loader would actually be using the new parser under the hood.

thanks for looking.

Reply:
I'll accept this as is, for now; we don't have public API besides the plugin interface, but I ask you to stick around for any regression bug reports and help out with that please.

Reply:
> I ask you to stick around for any regression bug reports and help out with that please.

100% please ping me if anything comes through that I don't notice and I'll jump on it.